### PR TITLE
GitHub Search Module

### PR DESCRIPTION
**Pull request by prince-7**
GitHub Search module (mentioned in issue #45)

### CODE REVIEW

File: modules/search/github.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import re
+
+class Module(BaseModule):
+
+	meta = {
+		'name': 'GitHub Search',
+		'author': 'Aman Singh',
+		'version': '0.1',
+		'description': 'Search your query in the GitHub and show the results.',
+		'sources': ('google', 'carrot2', 'bing', 'yippy'),
```
Comment (saeeddhqan) [inline modules/search/github.py]: use other sources such as qwant, millionshort and duckduckgo

File: modules/search/github.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        'examples': ('github -q <QUERY> -l 15 --output',)
+	}
+
+	def module_run(self):
+		query = self.options['query']
+		limit = self.options['limit']
+		count = self.options['count']
+		engine = self.options['engine'].split(',')
+		google_q = f"site:github.com {query}"
+		bing_q = f"site:github.com {query}"
```
Comment (saeeddhqan) [inline modules/search/github.py]: Don't define a variable three times. look at the Reddit module.

File: modules/search/github.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+					link = link.split('/')[3]
+				except:
+					continue
+				if re.search(r'^[\w\d_\-\/]+$', link):
+					if(link not in users):
+						users.append(link)
+						self.output(f"\t{link}", 'G')
+
+			self.alert('Repository')
+			for link in links:
```
Comment (saeeddhqan) [inline modules/search/github.py]: there's no need to show repositories. usernames and links are enough. But if you want to show it, show it with the username. not just the repo name.

**Comment by prince-7** [comment]
Updates are done, please check.

File: modules/search/github.py (hunk 3 above)
Comment (prince-7) [reply modules/search/github.py]: Added the username along with repo name

File: modules/search/github.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+			pages += run.pages
+			for item in run.links_with_title:
+				link,title = item
+				self.verbose(f"\t{title}", 'C')
+				self.verbose(f"\t\t{link}")
+				self.verbose('')
+				links.append(link)
+
+		if 'carrot2' in engine:
+			run = self.carrot2(google_q)
```
Comment (saeeddhqan) [inline modules/search/github.py]: we don't have google_q, yippy_q,...

File: modules/search/github.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+				if re.search(r'^[\w\d_\-\/]+$', link):
+					if(link not in repository):
+						repository.append(link)
+						self.output(f"\t{link}", 'G')
+
+			self.alert('Links')
+			for link in links:
+				self.output(f'\t{link}')
+
+		self.save_gather({'users': users, 'repo': repository,}, 'search/github', query, output=self.options.get('output'))
```
Comment (saeeddhqan) [inline modules/search/github.py]: remove "," after repository var

File: modules/search/github.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+		count = self.options['count']
+		engine = self.options['engine'].split(',')
+		q = f"site:github.com {query}"
+
+		run = self.google(q, limit, count)
+		run.run_crawl()
+		links = run.links
+		users = []
+		repository = []
+		pages = run.pages
```
Comment (saeeddhqan) [inline modules/search/github.py:49]: you didn't use it. You can use it to find blogs: check the youtube module

File: modules/search/github.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+import re
+
+class Module(BaseModule):
+
+	meta = {
+		'name': 'GitHub Search',
+		'author': 'Aman Singh',
+		'version': '0.1',
+		'description': 'Search your query in the GitHub and show the results.',
+		'sources': ('google', 'carrot2', 'bing', 'yippy', 'yahoo', 'millionshort','qwant'),
```
Comment (saeeddhqan) [inline modules/search/github.py]: add duckduckgo

File: modules/search/github.py (hunk 6 above)
Comment (prince-7) [reply modules/search/github.py:49]: Done

**Pull request merged**